server/controllers/userController.js

```diff
@@ -239,6 +239,103 @@ class UserController {
         return util.errorStatus(res, 500, err.message);
     }
   }
+
+  static async getUserDetails(req, res) {
+    const { userId } = req.query;
+    let data;
+
+    try {
+      if (userId) {
+        const theUser = await models.Users.findByPk(userId);
+
+        if (!theUser) {
+          return util.errorStatus(res, 404, 'User not found');
+        }
+        const { id, firstName, lastName, email, signupMethod } = theUser;
+        data = {
+          id,
+          firstName,
+          lastName,
+          email,
+          signupMethod
+        };
+      } else {
+        const theUser = await models.Users.findAll();
```

Comment:
You do not need to map through the objects to pick what you want. You can get that done from the query.
eg: 
```
Project.find({
  attributes: ['id', 'name', 'title']
})
```